ext/winevt/winevt_subscribe.c

```diff
@@ -144,17 +162,67 @@ rb_winevt_subscribe_subscribe(int argc, VALUE* argv, VALUE self)
   return Qfalse;
 }
 
+BOOL
+rate_limit_default_check_handler(struct WinevtSubscribe *winevtSubscribe)
+{
+  return FALSE;
+}
+
+void
+rate_limit_default_state_handler(struct WinevtSubscribe *winevtSubscribe,
+                                 ULONG count)
+{
+}
+
+BOOL
+rate_limit_check_handler(struct WinevtSubscribe *winevtSubscribe)
+{
+  time_t now;
+  ULONG currRate = 0;
+
+  time(&now);
+
+  winevtSubscribe->previousRate = 0;
```

Comment:
Is `previousRate` is needed?

Reply:
This line should be needed.
Otherwise, after exceeding rate limit, winevt_c processes every 10 events.
It is not expected behavior.

Reply:
I'm not sure but it seems that same feature can be implemented by resetting `currentRate` after `if` clause.
I'm sorry if it's wrong.

Reply:
I couldn't implement it with your suggested change.
And it is very getting confuse for me.

Reply:
Oh, `else` clause did the trick:
https://github.com/fluent-plugins-nursery/winevt_c/pull/9/commits/f8a754280933f5dc5cd8f05756c9729abe4b0bab